src/rgw/rgw_rados.cc

```diff
@@ -4476,12 +4476,12 @@ int RGWRados::transition_obj(RGWObjectCtx& obj_ctx,
 
 int RGWRados::check_bucket_empty(RGWBucketInfo& bucket_info, optional_yield y)
 {
-  std::vector<rgw_bucket_dir_entry> ent_list;
   rgw_obj_index_key marker;
   string prefix;
   bool is_truncated;
 
   do {
+    std::vector<rgw_bucket_dir_entry> ent_list;
```

Comment:
clear() is probably faster than reallocation of the vector

Reply:
That's certainly possible. In the interest of getting this and other fixes (built on top of this) in, though, I'm going to merge this. Feel free to follow up w/ another PR.